test/test.c

```diff
@@ -55,7 +55,8 @@ unsigned short test_generate_animal()
 	printf("Running test %s ... \n", __FUNCTION__);
 	unsigned short is_success = 1;
 	for (int i = 0; i < EXPERIMENTS_COUNT; i++) {
-		struct animal actual_data = generate_animal();
+		struct animal actual_data;
+		generate_animal(&actual_data);
 		if (strcmp(get_animal_type_name(actual_data.type), "N/A") ==
```

Comment:
```suggestion
		if (strcmp(get_animal_type_name(actual_data.type), expected_values[i]) == 0) {
```

please. 

Это был мой косяк